mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -365,7 +368,7 @@ private boolean userHasLookupRightsOn(Mailbox mailbox, MailboxSession session) t
     private List<MailboxId> performConcurrentMailboxCreation(MailboxSession mailboxSession, MailboxPath mailboxPath) throws MailboxException {
         List<MailboxId> mailboxIds = new ArrayList<>();
         locker.executeWithLock(mailboxPath, () ->
-            block(mailboxExists(mailboxPath, mailboxSession)
+            block(Mono.from(mailboxExists(mailboxPath, mailboxSession))
```

Comment:
Can we make `block` accept a `Publisher` to reduce boiler plate?

Reply:
i'll try